client/jetpack-connect/controller.js

```diff
@@ -66,6 +75,23 @@ const jetpackNewSiteSelector = context => {
 	);
 };
 
+const getPlanSlugFromFlowType = ( type, interval = 'yearly' ) => {
+	const planSlugs = {
+		yearly: {
+			personal: PLAN_JETPACK_PERSONAL,
+			premium: PLAN_JETPACK_PREMIUM,
+			pro: PLAN_JETPACK_BUSINESS,
+		},
+		monthly: {
+			personal: PLAN_JETPACK_PERSONAL_MONTHLY,
+			premium: PLAN_JETPACK_PREMIUM_MONTHLY,
+			pro: PLAN_JETPACK_BUSINESS_MONTHLY,
+		},
+	};
+
+	return planSlugs[ interval ][ type ] || '';
```

Comment:
I _believe_ this is safe with the way routes and defaults are defined, but I'd prefer to be more robust here using a util like `_.get`.